en/docs/learn/api-security/oauth2/grant-types/jwt-grant.md

```diff
@@ -1,69 +1,137 @@
+    ```
+
+Now you have obtained a JWT from WSO2 IS. In the next step, let us register an identity provider and create a service provider in WSO2 API-M.
+
+### Step 2 - Configure an Identity Provider and a Service Provider in WSO2 API-M
+
+1. Make sure WSO2 API-M is [up and running]({{base_path}}/install-and-setup/installation-guide/running-the-product/#starting-the-server).
+
+2. Sign in to the WSO2 API-M Management Console (`https://<API-M_Server_Host>:9443/carbon`)   
+
+3. Navigate to the **Identity Providers** section in the **Main** -> **Identity** section and click **Add.**
+
+4. Provide the following values to configure the Identity Provider.
+    -   **Identity Provider Name :** Enter the issuer name as the identity provider name. This is used to generate the JWT assertion. The issuer of a JWT generated by WSO2 IS is by default `https://localhost:9444/oauth2/token`.
+    -   **Identity Provider Public Certificate :** Choose the radio button named **Upload IDP certificate** so that the field named **Identity Provider Public Certificate** will be enabled. This is the certificate which will be used to sign the JWT assertion.
+
+        !!! info
+            The **Identity Provider Public Certificate** is the public certificate belonging to the identity provider. Uploading this is necessary to authenticate the response from the identity provider.
+
+            This can be any certificate. If the identity provider is another API Manager or an Identity Server, this can be a `wso2.crt` file. 
+            
+            Since we have used WSO2 IS as the identity provider here, follow the below steps to create the identity provider certificate from the `wso2carbon.jks file`.
+
+            - Open your command line interface, navigate to the `<IS_HOME>/repository/resources/security/` directory and run the following command.
+              ``` java
+              keytool -export -alias wso2carbon -file wso2.crt -keystore wso2carbon.jks -storepass wso2carbon
+              ```
 
-7.  Enter a **Callback URL.** For example, use `https://wso2is.local:8080/playground2/oauth2client` and click **Add**. The **OAuth Client Key** and **OAuth Client Secret** will now be visible.
+            - Once you run this command, the `wso2.crt` file will be generated inside the same folder. You can upload it by clicking **Choose File** in Identity Provider section in the Management Console.
+
+            See [Using Asymmetric Encryption](https://docs.wso2.com/display/ADMIN44x/Using+Asymmetric+Encryption) in the WSO2 Product Administration Guide for more information on how public keys work and how to sign these keys by a certification authority.
```

Comment:
This doc link is deprecated. Shall we point to the https://apim.docs.wso2.com/en/latest/install-and-setup/setup/security/configuring-keystores/keystore-basics/about-asymetric-cryptography/

Reply:
Fixed via https://github.com/wso2/docs-apim/pull/3170/commits/d99c31c64db63b3df9a1aad0bc9edde2c10379f0